From cc6197a382c3123186942e0a07c5ce3840e59472 Mon Sep 17 00:00:00 2001
From: Ravi Vembu <raviv@motorola.com>
Date: Wed, 18 Mar 2015 11:28:08 -0500
Subject: [PATCH 230/959] IKSWL-2981: power: qpnp-smbcharger-mmi: Skip APSD
 rerun for Factory mode

APSD re-run can be invoked if usb interrupts mess up the re_run count.
Skip APSD rerun entirely in factory mode.
Signed-off-by: Ravi Vembu <raviv@motorola.com>

Change-Id: I6b85fff246b3fdfb5418250c260e68b98d99058c
Reviewed-on: http://gerrit.mot.com/726205
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Ravikumar Vembu <raviv@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/power/qpnp-smbcharger-mmi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index d60fbbcb691..37bf6e48dea 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -3969,7 +3969,7 @@ static void handle_usb_insertion(struct smbchg_chip *chip)
 
 	/* Rerun APSD 1 sec later */
 	if ((usb_supply_type == POWER_SUPPLY_TYPE_USB) &&
-	    !chip->apsd_rerun_cnt) {
+	    !chip->apsd_rerun_cnt && !chip->factory_mode) {
 		dev_info(chip->dev, "HW Detected SDP!\n");
 		chip->apsd_rerun_cnt++;
 		chip->usb_present = 0;
-- 
2.11.0

